lib/dry/configurable/setting.rb

```diff
@@ -109,23 +114,20 @@ def writer?(meth)
         writer_name.equal?(meth)
       end
 
-      # @api private
-      def clonable_value?
-        CLONABLE_VALUE_TYPES.any? { |type| value.is_a?(type) }
-      end
-
       private
 
       # @api private
       def initialize_copy(source)
         super
-        @value = source.value.dup if source.input_defined? && source.clonable_value?
+        @input = source.input.dup if Setting.clonable_value?(source.input)
+        @default = source.default.dup if Setting.clonable_value?(source.default)
+        @value = source.value.dup if source.evaluated? && Setting.clonable_value?(source.value)
```

Comment:
@esparta Originally I didn't have the "clonable value?" postfix conditional around the @value assignment here, but then I double checked the diff and realised they were in place before, i.e.

```ruby
@value = source.value.dup if source.input_defined? && source.clonable_value?
```

So I thought it would be best to preserve this aspect of the previous behaviour.